src/main/java/duke/subfiles/TaskList.java

```diff
@@ -0,0 +1,370 @@
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Task;
+import duke.task.ToDo;
+
+/**
+ * The TaskList class contains a list of tasks created by
+ * user input, and allows the user to add, print, or delete
+ * tasks, as well as to mark a task in the list as done.
+ *
+ * @author  arsatis
+ * @version 1.1
+ * @since   2021-01-26
+ */
+public class TaskList {
+    /** List of tasks created by user input. */
+    private ArrayList<Task> tasks;
+
+    /**
+     * Default constructor for the TaskList class.
+     */
+    public TaskList() {
+        tasks = new ArrayList<>();
+    }
+
+    /**
+     * Adds a to-do to the list of tasks.
+     *
+     * @param s User input triggering the addition of a to-do
+     *          to the list of tasks.
+     * @throws EmptyDescriptionException If no description is provided
+     *                                   for the to-do.
+     */
+    private void addTodo(String s)
+            throws EmptyDescriptionException {
+        try {
+            s = s.substring(5);
+            tasks.add(new ToDo(s));
+        } catch (StringIndexOutOfBoundsException e) {
+            throw new EmptyDescriptionException("todo");
+        }
+    }
+
+    /**
+     * Adds a deadline to the list of tasks.
+     *
+     * @param s User input triggering the addition of a deadline
+     *          to the list of tasks.
+     * @throws EmptyDescriptionException If no description is provided
+     *                                   for the deadline.
+     * @throws EmptyDateException If no date is specified for the deadline.
+     * @throws DateFormatException If the specified date is incorrectly
+     *                             formatted.
+     */
+    private void addDeadline(String s)
+            throws EmptyDescriptionException, EmptyDateException, DateFormatException {
+        try {
+            String[] sArray = s.split("/", 2);
+            s = sArray[0].substring(9, sArray[0].length() - 1);
+            LocalDate t = LocalDate.parse(sArray[1].substring(3));
+            tasks.add(new Deadline(s, t));
+        } catch (StringIndexOutOfBoundsException e) {
+            throw new EmptyDescriptionException("deadline");
+        } catch (ArrayIndexOutOfBoundsException e) {
+            throw new EmptyDateException("deadline");
+        } catch (DateTimeParseException e) {
+            throw new DateFormatException();
+        }
+    }
+
+    /**
+     * Adds an event to the list of tasks.
+     *
+     * @param s User input triggering the addition of an event
+     *          to the list of tasks.
+     * @throws EmptyDescriptionException If no description is provided
+     *                                   for the event.
+     * @throws EmptyDateException If no date is specified for the event.
+     * @throws DateFormatException If the specified date is incorrectly
+     *                             formatted.
+     */
+    private void addEvent(String s)

```

Comment:
Will it be better if you change the parameter name s to a more meaningful word like "taskName"?